RecoEgamma/EgammaTools/src/GBRForestTools.cc

```diff
@@ -27,13 +27,30 @@ namespace {
 
 };
 
-std::unique_ptr<const GBRForest> GBRForestTools::createGBRForest(const std::string &weightFile){
-  edm::FileInPath weightFileEdm(weightFile);
-  return GBRForestTools::createGBRForest(weightFileEdm);
+// Creates a pointer to new GBRForest corresponding to a TMVA weights file,
+// given a file path as string.  When the file is not found, also the same name
+// appended with ".gz" and ".gzip" is tried.  This allows the maintainters of
+// the data repositories to gzip weight files a posteriori without changing
+// anything else.
+std::unique_ptr<const GBRForest> GBRForestTools::createGBRForest(const std::string &weightFile,
+                                                                 std::vector<std::string> &varNames){
```

Comment:
I'm missing the reason why the ```varNames``` are needed.

Reply:
To automatically retrieve the variables needed by the BDT by name in the weight file. In the past, the variable preparation (in the right order) was always hardcoded in C++ and we want to change that (as I prepare here https://github.com/guitargeek/cmssw/blob/ElectronID_MVA2017_V2_1020pre3/RecoEgamma/ElectronIdentification/src/ElectronMVAEstimatorRun2.cc#L59).

This will be used in another egamma PR to come. I just did it already now so we don't have to touch the GBRForest tools again later. But as you like, I can also undo it for now!

Reply:
OK, thanks for the clarification.